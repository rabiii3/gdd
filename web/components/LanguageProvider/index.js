import { compose } from 'recompose';
import { connect } from 'react-redux';
import component from './component';
import { getLocale } from '../../selectors/language';

const mapStateToProps = state => ({
  locale: getLocale(state),
});

export const enhance = compose(connect(mapStateToProps));

export default enhance(component);
