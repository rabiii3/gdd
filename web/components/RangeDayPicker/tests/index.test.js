import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import LanguageProvider from '../../../components/LanguageProvider';
import configureStore from '../../../store/configureStore.test';
import { enhance } from '../';

describe('web | components | RangeDayPicker | index', () => {
  describe('render', () => {
    it('should render', () => {
      const Component = () => <div />;
      const EnhancedComponent = enhance(Component);
      const state = { language: { locale: 'en' } };
      const store = configureStore(state, null);

      const wrapper = mount(
        <Provider store={store}>
          <LanguageProvider messages={{}}>
              <EnhancedComponent />
          </LanguageProvider>
        </Provider>,
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
