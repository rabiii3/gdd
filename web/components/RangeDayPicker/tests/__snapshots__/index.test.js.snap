// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`web | components | RangeDayPicker | index render should render 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(LanguageProvider)
    messages={Object {}}
  >
    <LanguageProvider
      dispatch={[Function]}
      locale="en"
      messages={Object {}}
    >
      <IntlProvider
        key="en"
        locale="en"
      >
        <WithStyles(Connect(withState(withHandlers(Component))))>
          <Connect(withState(withHandlers(Component)))
            classes={
              Object {
                "dayPicker": "Connect-withState-withHandlers-Component----dayPicker-1",
                "padding": "Connect-withState-withHandlers-Component----padding-2",
              }
            }
          >
            <withState(withHandlers(Component))
              classes={
                Object {
                  "dayPicker": "Connect-withState-withHandlers-Component----dayPicker-1",
                  "padding": "Connect-withState-withHandlers-Component----padding-2",
                }
              }
              handleStatisticsDates={[Function]}
              initialState={
                Object {
                  "from": 2019-05-31T23:00:00.000Z,
                  "to": 2019-08-30T23:00:00.000Z,
                }
              }
            >
              <withHandlers(Component)
                classes={
                  Object {
                    "dayPicker": "Connect-withState-withHandlers-Component----dayPicker-1",
                    "padding": "Connect-withState-withHandlers-Component----padding-2",
                  }
                }
                filter={null}
                handleFilter={[Function]}
                handleStatisticsDates={[Function]}
                initialState={
                  Object {
                    "from": 2019-05-31T23:00:00.000Z,
                    "to": 2019-08-30T23:00:00.000Z,
                  }
                }
              >
                <Component
                  classes={
                    Object {
                      "dayPicker": "Connect-withState-withHandlers-Component----dayPicker-1",
                      "padding": "Connect-withState-withHandlers-Component----padding-2",
                    }
                  }
                  filter={null}
                  handleFilter={[Function]}
                  handleStatisticsDates={[Function]}
                  initialState={
                    Object {
                      "from": 2019-05-31T23:00:00.000Z,
                      "to": 2019-08-30T23:00:00.000Z,
                    }
                  }
                  setCloseFilter={[Function]}
                  setShowFilter={[Function]}
                >
                  <div />
                </Component>
              </withHandlers(Component)>
            </withState(withHandlers(Component))>
          </Connect(withState(withHandlers(Component)))>
        </WithStyles(Connect(withState(withHandlers(Component))))>
      </IntlProvider>
    </LanguageProvider>
  </Connect(LanguageProvider)>
</Provider>
`;
